packages/fuels-contract/src/contract.rs

```diff
@@ -45,6 +43,57 @@ pub struct CallResponse<D> {
     pub receipts: Vec<Receipt>,
 }
 
+pub struct ContractCallArguments {
```

Comment:
Ah this explains my confusion. Your change is to the internals rather than the user-facing interface. Builder pattern is much more useful for user-facing things. Internally I think it could even be counter-productive since you _want_ compiler errors when you change things, rather an unexpected behavior at runtime.

```suggestion
struct ContractCallArguments {
```

Reply:
Ok so that's why I was confused by this issue. Actually we already  have builder methods for the _user-facing_ call method, see an example of that here:https://github.com/FuelLabs/fuels-rs/blob/5edde493227459246eeec937d1a4855b7f8a40b2/packages/fuels-abigen-macro/tests/harness.rs#L619

Reply:
Can we close this PR and issue then? I think the original suggestion to use a builder pattern was also assuming this was user-facing due to `pub`.

Reply:
I think so yes.